### PR TITLE
map GeoNames feature codes to the Pelias categories taxonomy (#20)

### LINKED ISSUE #20: map FeatureClass and FeatureCode to the pelias taxonomy
Using [feature Class](http://www.geonames.org/source-code/javadoc/org/geonames/FeatureClass.html) and/or [feature code](http://www.geonames.org/export/codes.html) we can categorize geonames 

* featureClass ```A``` is an admin boundary for instance
* featureClass ```P``` is a populated place

Based on the featureClass, we have a bunch of featureCodes we can make use of .. for example:

* ```REST```: featureCode that denotes a given point is a restaurant
* ```AIRP```: Airport
* ```BLDG```: Building

We could use this in conjunction to https://github.com/pelias/openstreetmap/issues/33

### PR DESCRIPTION
**Pull request by sevko**
Map the extensive [geonames feature codes](http://www.geonames.org/export/codes.html) onto the Pelias taxonomy.

  * add a category mappings file, `metadata/category_mapping.json` (mostly uses the same keys as [openstreetmap](https://github.com/pelias/openstreetmap/blob/master/config/category_map.js), but there are some new ones like `admin`, `industry`, `academic`, `natural:water`)
  * set `Document` categories in `task/import.js` 

fixes #20

### CODE REVIEW

**Comment by missinglink** [comment]
what do you see as the semantic difference between `academic` and `education`?

**Comment by sevko** [comment]
Looks like I missed `education` when looking that file over, my fault. `academic` sounds like a better fit since it's more general (ie, a lab might be classified as academic but not necessarily educational), but seeing as it's already used in OSM I'm fine with leaving it as is, and changing all these occurrences of `academic` to `education`.

I departed from the OSM schema in some places, introducing categories like `natural` and `natural:water`, because what we currently have is a bit restrictive. It makes more sense to categorize a marsh, for instance, as simply a natural, watery area rather than a recreational one. I also added tags for `industry`, `industry:agriculture`, etc.

**Comment by missinglink** [comment]
The idea is to provide a unified taxonomy, so things like https://github.com/pelias/openstreetmap/blob/master/config/category_map.js#L264 need to both map to the same pelias category.

In order to be consistent we need to either change the `pelias/openstreetmap` categories if they are wrong or change these ones if they are wrong, but we cannot have the same concept mapping to different categories.

**Comment by sevko** [comment]
Totally agree, we need to discuss the new categories before this gets merged.

**Comment by sevko** [comment]
As per [these changes to the taxonomy](https://github.com/pelias/pelias/wiki/Taxonomy-v1/_compare/eca8df97503721c8e73b98897b35dc264b7f8860...7616b793f0c564011ee6953aa2479cd5ce9542e1), we've decided to add top-level keys `industry` and `natural`. We'll update the `pelias/openstreetmap` mapping accordingly. There are also several issues in this PR that I'll need to address:

  * there's a mix of `agriculture` and `industry:agriculture` categories; only use `industry:agriculture`
  * use the `finance` category where it makes sense
  * use `education` instead of `academic`

**Comment by missinglink** [comment]
:+1:

**Comment by hkrishna** [comment]
looks good :+1: :shipit:

File: task/import.js [hunk 1]
```diff
@@ -47,6 +48,18 @@ function mapper( data, enc, next ){
...
     } catch( err ){}
 
+    if( typeof data.feature_code === 'string' ){
+      var featureCode = data.feature_code.toUpperCase();
+      var peliasCategories = categoryMapping[ featureCode ];
+      if( peliasCategories !== undefined ){
+        peliasCategories.forEach( function ( category ){
+          try {
+            record.addCategory( category );
+          } catch ( ex ) {}
```
Comment (dianashk) [inline task/import.js]: should probably log this error

File: task/import.js [hunk 2]
```diff
@@ -47,6 +48,18 @@ function mapper( data, enc, next ){
       }
     } catch( err ){}
 
+    if( typeof data.feature_code === 'string' ){
+      var featureCode = data.feature_code.toUpperCase();
+      var peliasCategories = categoryMapping[ featureCode ];
```
Comment (dianashk) [inline task/import.js]: could use hasOwnProperty

**Comment by dianashk** [comment]
:+1:

**Pull request merged**